### PR TITLE
Preview mode

### PR DESCRIPTION
**Pull request by ismaeldcom**
## PR Checklist

Please check all that apply to this PR using "x":

- [x] I have checked that this PR is not a duplicate of an existing PR (open, closed or merged)
- [x] I have checked that this PR does not introduce a breaking change
- [ ] This PR introduces breaking changes and I have provided a detailed explanation below

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting)
- [ ] Refactoring (no functional changes)
- [ ] Documentation changes
- [ ] Other - Please describe:

## Fixes

Issue Number: #27 

## What is the current behavior?
None
## What is the new behavior?
Preview mode that load only first 3 images and show a label on top right corner
## Other information
I always like to start with the simplest approach, If you like I can make preview size configurable, add a placeholder for "missing" pages...

### CODE REVIEW

File: src/components/uncompress.js [hunk 1]
```diff
@@ -79,12 +79,14 @@ class Uncompress extends Component {
   }
 
   extract = async blob => {
+    const { preview } = this.context.state
```
Comment (btzr-io) [inline src/components/uncompress.js]: this should be a `prop` of the component it self

File: src/components/uncompress.js [hunk 2]
```diff
@@ -79,12 +79,14 @@ class Uncompress extends Component {
...
     try {
       // Compressed files 1437
       const list = await this.openArchive(blob)
 
       if (list && list.length > 0) {
-        await asyncForEach(list, async (item, index) => {
+        const items = preview ? list.splice(0,3) : list
```
Comment (btzr-io) [inline src/components/uncompress.js]: The limit should be configurable, also probably a `prop`

File: src/css/styles.css [hunk 3]
```diff
@@ -63,6 +63,18 @@
   fill: var(--icon-color);
 }
 
+.villain-preview::after {
```
Comment (btzr-io) [inline src/css/styles.css]: this should be a component or element not a `pseudo-element`

**Comment by btzr-io** [comment]
@ismaeldcom can you add a screenshot ?

**Comment by ismaeldcom** [comment]
Ok, I'll take note of these comments, here is a screenshot
preview

File: src/components/uncompress.js (hunk 2 above)
Comment (ismaeldcom) [reply src/components/uncompress.js]: You prefer 2 props like preview & previewCount or only one that active preview if value is greater than 1?

**Comment by btzr-io** [comment]
@ismaeldcom Nice thanks, looks good :+1:

It shows 6 pages ?

**Comment by ismaeldcom** [comment]
Yes, I should explain that. I thought was a good idea to show the total pages even if you can only preview N. This is why I asked about a placeholder to show in the missing pages. You have another idea?

**Comment by btzr-io** [comment]
@ismaeldcom  You need to update / set `totalPages` to the new limit.
https://github.com/btzr-io/Villain/blob/b00ba2443d88702280470cf2f57f04ea7018e765/src/context.js#L10

**Comment by ismaeldcom** [comment]
Yes I know where to set the total pages, I just let it the full length because of my last comment. But I'll change it.

**Comment by ismaeldcom** [comment]
Maybe you didn't see the comment about the 2 props:
`You prefer 2 props like preview & previewCount or only one that active preview if value is greater than 1?`

What do you think?

**Comment by btzr-io** [comment]
@ismaeldcom I agree with showing the full pages but that will break the toolbar, should probably show it somewhere else, anyway not a priority right now.

**Comment by btzr-io** [comment]
@ismaeldcom yeah just one prop is fine :+1:

**Comment by ismaeldcom** [comment]
Ok, I'll show the preview count for now.
To avoid toolbar break an option may be show the preview pages and fill in the rest with *placeholder* images or a message like *You're looking a preview, please load the full version to continue* or something like that. But is just an idea, I don't know your thoughts 😉

**Comment by btzr-io** [comment]
@ismaeldcom I think will be better to show one single message or place holder like:
```
 - Showing 3 pages of 45
-  You are reading a preview version, to access to full version do something.... ```
```
 But think that's up to the `dev` implementing this reader on a website or project, since this can be used for multiple proposes.

**Comment by btzr-io** [comment]
Great this is ready to be merged, thanks :v:

Could you add `preview` to readme?
https://github.com/btzr-io/Villain/blob/master/README.md#options

If you don't want I'll do it :+1:

**Comment by ismaeldcom** [comment]
@btzr-io Done 👍

**Pull request merged**